api/src/main/java/org/openmrs/Obs.java

```diff
@@ -1177,91 +1171,13 @@ public void setDateCreated(Date dateCreated) {
-	 * <strong>Should</strong> return null if the path is not specified
-	 */
-	public String getFormFieldPath() {
-		if (StringUtils.isNotBlank(formNamespaceAndPath)) {
-			//Only the namespace was specified
-			if (formNamespaceAndPath.endsWith(FORM_NAMESPACE_PATH_SEPARATOR)) {
-				return null;
-			}
-			return formNamespaceAndPath.substring(formNamespaceAndPath.indexOf(FORM_NAMESPACE_PATH_SEPARATOR) + 1);
-		}
-		
-		return formNamespaceAndPath;
-	}
-	
-	/**
-	 * Sets the namespace and path of the form field that was used to capture the obs details in the
-	 * form.<br>
-	 * <b>Note:</b> Namespace and formFieldPath together must not exceed 254 characters in length,
-	 * form applications can subtract the length of their namespace from 254 to determine the
-	 * maximum length they can use for a form field path.
-	 * 
-	 * @param namespace the namespace of the form field
-	 * @param formFieldPath the path of the form field
-	 * @since 1.11
-	 * <strong>Should</strong> set the underlying formNamespaceAndPath in the correct pattern
-	 * <strong>Should</strong> reject a namepace containing the separator
-	 * <strong>Should</strong> reject a path containing the separator
-	 * <strong>Should</strong> reject a namepace and path combination longer than the max length
-	 * <strong>Should</strong> not mark the obs as dirty when the value has not been changed
-	 * <strong>Should</strong> mark the obs as dirty when the value has been changed
-	 * <strong>Should</strong> mark the obs as dirty when the value is changed from a null to a non null value
-	 * <strong>Should</strong> mark the obs as dirty when the value is changed from a non null to a null value
+	 * @see org.openmrs.FormRecordable#setFormField(String,String)
 	 */
+	@Override
 	public void setFormField(String namespace, String formFieldPath) {
-		if (namespace == null && formFieldPath == null) {
-			markAsDirty(formNamespaceAndPath, null);
-			formNamespaceAndPath = null;
-			return;
-		}
-		
-		String nsAndPathTemp = "";
-		if (StringUtils.isNotBlank(namespace) && StringUtils.isNotBlank(formFieldPath)) {
-			nsAndPathTemp = namespace + FORM_NAMESPACE_PATH_SEPARATOR + formFieldPath;
-		} else if (StringUtils.isNotBlank(namespace)) {
-			nsAndPathTemp = namespace + FORM_NAMESPACE_PATH_SEPARATOR;
-		} else if (StringUtils.isNotBlank(formFieldPath)) {
-			nsAndPathTemp = FORM_NAMESPACE_PATH_SEPARATOR + formFieldPath;
-		}
-		
-		if (nsAndPathTemp.length() > FORM_NAMESPACE_PATH_MAX_LENGTH) {
-			throw new APIException("Obs.namespaceAndPathTooLong", (Object[]) null);
-		}
-		if (StringUtils.countMatches(nsAndPathTemp, FORM_NAMESPACE_PATH_SEPARATOR) > 1) {
-			throw new APIException("Obs.namespaceAndPathNotContainSeparator", (Object[]) null);
-		}
-		
-		markAsDirty(this.formNamespaceAndPath, nsAndPathTemp);
-		formNamespaceAndPath = nsAndPathTemp;
+		String oldValue = formNamespaceAndPath;
+		super.setFormField(namespace, formFieldPath);
+		markAsDirty(oldValue, formNamespaceAndPath);
```

Comment:
Are you sure it's 100% ok, unfortunately there is also [this piece](https://github.com/openmrs/openmrs-core/blob/de7bfbb8f84ccf82ea5fd4e934c7a9ba9e483f31/api/src/main/java/org/openmrs/Obs.java#L1242):
```java
if (namespace == null && formFieldPath == null) {
  markAsDirty(formNamespaceAndPath, null);
  formNamespaceAndPath = null;
  return;
}
```

Reply:
I think this new code still takes care of that use case properly... ie marking as dirty if the formNamespaceAndPath goes from something to null, if I'm understanding it correctly.